multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/ProcessDescriptorStep.java

```diff
@@ -119,28 +132,34 @@ private void setDynamicResolvableParametersIfAbsent(ProcessContext context, MtaD
         }
     }
 
+    private List<String> findInvalidModulesSpecifiedForDeployment(DeploymentDescriptor descriptor, List<String> modulesForDeployment) {
+        if (CollectionUtils.isEmpty(modulesForDeployment)) {
+            return Collections.emptyList();
+        }
+        Set<String> deploymentDescriptorModuleNames = descriptor.getModules()
+                                                                .stream()
+                                                                .map(Module::getName)
+                                                                .collect(Collectors.toSet());
+        return modulesForDeployment.stream()
+                                   .filter(moduleSpecifiedForDeployment -> !deploymentDescriptorModuleNames.contains(
+                                       moduleSpecifiedForDeployment))
+                                   .collect(Collectors.toList());
+    }
+
     private Set<String> getModuleNames(DeploymentDescriptor deploymentDescriptor, List<String> moduleNamesForDeployment) {
```

Comment:
maybe rename to getModuleNamesForDeployment or similar; otherwise it's a bit random when the flag is taken into account